pkg/resource/logging-agents-toggle/observability_bundle_configmap.go

```diff
@@ -27,9 +27,26 @@ func GenerateObservabilityBundleConfigMap(lc loggedcluster.Interface, observabil
 	if observabilityBundleVersion.LT(semver.MustParse("1.0.0")) {
 		promtailAppName = "promtail-app"
 	}
-	appsToEnable[promtailAppName] = app{
-		Enabled: true,
+
+	switch lc.GetLoggingAgent() {
+	case "promtail":
```

Comment:
This is missing a check if the observability bundle even has alloy

Reply:
i would rather we default to promtail if it's not there (checking olly bundle version)

Reply:
something like `observabilityBundleVersion.GE(semver.MustParse("1.4.0")` ?

Reply:
Yes that would be nice :)

Reply:
I added some logic within the `logging-agent-toggle` resource, to select the correct logging-agent. The info is then persisted within the `loggedcluster.Interface` object. This avoid to duplicate the logic to fetch and compare the observability-bundle version.